app/controllers/products_controller.rb

```diff
@@ -37,6 +39,22 @@ def destroy
     end
   end
 
+  def add_to_cart
+    if logged_in?
+      id = params[:id].to_i
+      session[:cart] << id unless session[:cart].include?(id)
+      redirect_to root_path
+    else
+      redirect_to login_path
+    end
```

Comment:
This can also be written like:

```
return redirect_to login_path unless logged_in?

session[:cart] << id unless session[:cart].include?(id)
redirect_to root_path
```

I used the method `id` which I mentioned in the comment below